src/msw/treectrl.cpp

```diff
@@ -1608,54 +1608,34 @@ void wxTreeCtrl::Delete(const wxTreeItemId& item)
     // tree ctrl will eventually crash after item deletion
     TreeItemUnlocker unlock_all;
 
-    if ( HasFlag(wxTR_MULTIPLE) )
-    {
-        bool selected = IsSelected(item);
-        wxTreeItemId next;
-
-        if ( selected )
-        {
-            next = TreeView_GetNextVisible(GetHwnd(), HITEM(item));
-
-            if ( !next.IsOk() )
-            {
-                next = TreeView_GetPrevVisible(GetHwnd(), HITEM(item));
-            }
-        }
+    bool selected = IsSelected(item);
 
-        if ( !MSWDeleteItem(item) )
-            return;
-
-        if ( !selected )
-        {
-            return;
-        }
+    if ( !MSWDeleteItem(item) || !selected )
```

Comment:
This condition is not really obvious to read, I'd use two `if`s here, one checking for the failure to delete the item at all and another checking if we need to generate the selection changing events (with an appropriate comment).